### ISSUE TITLE
slug property is not correct when using directory and index.md approach

### ISSUE THREAD

**Issue by davydnorris** [nuxt/content#385]
### Version
@nuxt/content: 1.6.0
nuxt: 2.14.1

### Reproduction Link
https://codesandbox.io/s/dank-paper-djlm0?file=/content/a-new-entry/index.md

### Steps to reproduce
Create two content files
 - one with a unique name at the top level in the content directory
 - one in a subdirectory, and name it index.md

Start the application and check the content. Supply the file name or the directory name as the slug in ` await $content(slug)`

### What is Expected?
the slug property on the returned post is the same as what had to be supplied in order to fetch it

### What is actually happening?
the slug property is correct for file names, but is being returned as 'index' for directory based content ie. the .md file name not the actual slug name

**Comment by benjamincanac** [comment]
@davydnorris The arguments of the `$content()` function are `paths`, not `slugs`. You have to specify to full path of a file to get it. The slug property of each file is its filename.

https://content.nuxtjs.org/fetching#contentpath-options

**Comment by davydnorris** [comment]
OK so there's maybe two things here.

I now see that you supply a path to $content, which makes sense.

But a slug is defined as "the unique identifying part of a web address, typically at the end of the URL", which is not necessarily the file name. In the case of this issue, it's the final part of the URL, which is either the file name (minus .md) in the case of discrete files, or the name of the enclosing directory in the case of an index.md.

Returning 'index' as the slug every single time is not useful, and not correct from an SEO and web point of view.

**Comment by azrikahar** [comment]
Just wondering, is the original issue being not able to view the `/a-new-entry` page? If it is, it's because when it's a directory, `$content('a-new-entry')` ends up returning an array, not an object. This means the `page` in `<nuxt-content :document="page" />` is actually an array, so it doesn't work properly. I modified it and made it work at this CodeSandbox: https://codesandbox.io/s/zen-morse-ronz9?file=/pages/_.vue

If you already knew it's an array and just wanted to highlight the slug issue here, do ignore my tangent above haha.

---

Regarding the slug issue, I noticed that currently **the slug is built by grabbing the last item after splitting the path into an array,** as seen here:

https://github.com/nuxt/content/blob/157262cc932b9644c5c0747ea79a361ededb60a5/packages/content/lib/database.js#L193-L194

https://github.com/nuxt/content/blob/157262cc932b9644c5c0747ea79a361ededb60a5/packages/content/lib/database.js#L210

which is why all `index.md` file will only get the slug `index` regardless the path they are in.

A fix that I can think of is by using regex to detect 2 criteria:
- the last part of the path is `/index`
- it must be a nested `/index`, such as `/***/index`, `/***/***/index` and so on

If it fits the criteria, we will then grab the **second last item** of the splitted array, Here are screenshots of result to show the comparison of the original implementation vs the proposed one with regex check.

| Original | Proposed |
| --- | --- |
| ![chrome_hBLQ0HnkNy](https://user-images.githubusercontent.com/42867097/91642664-c5d3b480-ea5f-11ea-8c36-884d9e773aab.png) | ![chrome_g5ltSGDqgN](https://user-images.githubusercontent.com/42867097/91642669-cec48600-ea5f-11ea-9d43-2156a827aa60.png) |

We can also see that any pages under `a-new-entry` will still work as usual and have the correct slug.

With that said, I am also wondering when are we able to actually use the slug because when we run `nuxt generate`, assuming we are using the approach like I'm using in the codesandbox above, we never need the slug since everything should be generated correctly. The final static site url will still be `https:/domain.com/a-new-entry`.

**Comment by davydnorris** [comment]
Thanks @azrikahar but no - I actually discovered the issue when I was trying to use the slug as an ID for comments, views etc on my blog. Since the slug should actually be the unique identifier for a page, it makes sense to use this as the ID for any saved comments etc. but then all of mine were returning 'index'!

I have worked around the bug in much the same way as you have above - immediately after fetching the pages, I have a line that sets the slug property for each post to the correct value by parsing the dir property as you did above. Then the rest of my code continues to use the slug property as a unique ID to store the attached comments.
